// @lobehub/ui/es/ChatInputArea/mobile/SendButton.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["loading", "onStop", "onSend"];
import { useTheme } from 'antd-style';
import { Loader2, SendHorizontal } from 'lucide-react';
import { readableColor } from 'polished';
import { memo } from 'react';
import ActionIcon from "../../ActionIcon";
import { jsx as _jsx } from "react/jsx-runtime";
var MobileChatSendButton = /*#__PURE__*/memo(function (_ref) {
  var loading = _ref.loading,
    onStop = _ref.onStop,
    onSend = _ref.onSend,
    rest = _objectWithoutProperties(_ref, _excluded);
  var theme = useTheme();
  var size = {
    blockSize: 36,
    fontSize: 16
  };
  return loading ? /*#__PURE__*/_jsx(ActionIcon, _objectSpread({
    active: true,
    icon: Loader2,
    onClick: onStop,
    size: size,
    spin: true
  }, rest)) : /*#__PURE__*/_jsx(ActionIcon, _objectSpread({
    icon: SendHorizontal,
    onClick: onSend,
    size: size,
    style: {
      background: theme.colorPrimary,
      color: readableColor(theme.colorPrimary),
      flex: 'none'
    }
  }, rest));
});
export default MobileChatSendButton;
