// @lobehub/ui/es/Highlighter/index.js
'use client';

import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["fullFeatured", "copyButtonSize", "children", "language", "className", "style", "copyable", "showLanguage", "type", "wrap", "bodyRender", "actionsRender"];
import { memo } from 'react';
import { Flexbox } from 'react-layout-kit';
import CopyButton from "../CopyButton";
import Tag from "../Tag";
import FullFeatured from "./FullFeatured";
import SyntaxHighlighter from "./SyntaxHighlighter";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var Highlighter = /*#__PURE__*/memo(function (_ref) {
  var fullFeatured = _ref.fullFeatured,
    _ref$copyButtonSize = _ref.copyButtonSize,
    copyButtonSize = _ref$copyButtonSize === void 0 ? 'site' : _ref$copyButtonSize,
    children = _ref.children,
    _ref$language = _ref.language,
    language = _ref$language === void 0 ? 'markdown' : _ref$language,
    className = _ref.className,
    style = _ref.style,
    _ref$copyable = _ref.copyable,
    copyable = _ref$copyable === void 0 ? true : _ref$copyable,
    _ref$showLanguage = _ref.showLanguage,
    showLanguage = _ref$showLanguage === void 0 ? true : _ref$showLanguage,
    _ref$type = _ref.type,
    type = _ref$type === void 0 ? 'block' : _ref$type,
    wrap = _ref.wrap,
    bodyRender = _ref.bodyRender,
    actionsRender = _ref.actionsRender,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useStyles = useStyles(type),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  var originalActions = copyable && /*#__PURE__*/_jsx(CopyButton, {
    content: children,
    placement: "left",
    size: copyButtonSize
  });
  var actions = actionsRender ? actionsRender({
    actionIconSize: copyButtonSize,
    content: children,
    language: language,
    originalNode: originalActions
  }) : originalActions;
  var originalBody = /*#__PURE__*/_jsx(SyntaxHighlighter, {
    language: language === null || language === void 0 ? void 0 : language.toLowerCase(),
    children: children
  });
  var body = bodyRender ? bodyRender({
    content: children,
    language: language,
    originalNode: originalBody
  }) : originalBody;
  if (fullFeatured) return /*#__PURE__*/_jsx(FullFeatured, _objectSpread(_objectSpread({
    actionsRender: actionsRender,
    bodyRender: bodyRender,
    copyable: copyable,
    language: language,
    showLanguage: showLanguage,
    type: type,
    wrap: wrap
  }, rest), {}, {
    children: children
  }));
  return /*#__PURE__*/_jsxs("div", _objectSpread(_objectSpread({
    className: cx(styles.container, !wrap && styles.nowrap, className),
    "data-code-type": "highlighter",
    style: style
  }, rest), {}, {
    children: [/*#__PURE__*/_jsx(Flexbox, {
      align: 'center',
      className: styles.button,
      flex: 'none',
      gap: 4,
      horizontal: true,
      children: actions
    }), showLanguage && language && /*#__PURE__*/_jsx(Tag, {
      className: styles.lang,
      children: language.toLowerCase()
    }), /*#__PURE__*/_jsx("div", {
      className: styles.scroller,
      children: body
    })]
  }));
});
export default Highlighter;
export { default as SyntaxHighlighter } from "./SyntaxHighlighter";
