// @lobehub/ui/es/Mermaid/FullFeatured.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["showLanguage", "content", "children", "className", "copyable", "actionsRender", "style"];
import { ChevronDown, ChevronRight } from 'lucide-react';
import { memo, useState } from 'react';
import { Flexbox } from 'react-layout-kit';
import ActionIcon from "../ActionIcon";
import CopyButton from "../CopyButton";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var MermaidFullFeatured = /*#__PURE__*/memo(function (_ref) {
  var showLanguage = _ref.showLanguage,
    content = _ref.content,
    children = _ref.children,
    className = _ref.className,
    copyable = _ref.copyable,
    actionsRender = _ref.actionsRender,
    style = _ref.style,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useState = useState(true),
    _useState2 = _slicedToArray(_useState, 2),
    expand = _useState2[0],
    setExpand = _useState2[1];
  var _useStyles = useStyles('block'),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  var container = cx(styles.container, className);
  var size = {
    blockSize: 24,
    fontSize: 14,
    strokeWidth: 2
  };
  var originalActions = copyable && /*#__PURE__*/_jsx(CopyButton, {
    content: content,
    placement: "left",
    size: size
  });
  var actions = actionsRender ? actionsRender({
    actionIconSize: size,
    content: content,
    originalNode: originalActions
  }) : originalActions;
  return /*#__PURE__*/_jsxs("div", _objectSpread(_objectSpread({
    className: container,
    "data-code-type": "mermaid",
    style: style
  }, rest), {}, {
    children: [/*#__PURE__*/_jsxs(Flexbox, {
      align: 'center',
      className: styles.header,
      horizontal: true,
      justify: 'space-between',
      children: [/*#__PURE__*/_jsx(ActionIcon, {
        icon: expand ? ChevronDown : ChevronRight,
        onClick: function onClick() {
          return setExpand(!expand);
        },
        size: {
          blockSize: 24,
          fontSize: 14,
          strokeWidth: 3
        }
      }), showLanguage && /*#__PURE__*/_jsx(Flexbox, {
        align: 'center',
        className: styles.select,
        gap: 2,
        horizontal: true,
        justify: 'center',
        children: /*#__PURE__*/_jsx("span", {
          children: "mermaid"
        })
      }), /*#__PURE__*/_jsx(Flexbox, {
        align: 'center',
        flex: 'none',
        gap: 4,
        horizontal: true,
        children: actions
      })]
    }), /*#__PURE__*/_jsx("div", {
      style: expand ? {} : {
        height: 0,
        overflow: 'hidden'
      },
      children: children
    })]
  }));
});
export default MermaidFullFeatured;
